server/src/main/java/org/elasticsearch/index/fielddata/IndexFieldDataService.java

```diff
@@ -111,7 +111,7 @@ public synchronized void clearField(final String fieldName) {
 
     @SuppressWarnings("unchecked")
     public <IFD extends IndexFieldData<?>> IFD getForField(MappedFieldType fieldType, String fullyQualifiedIndexName) {
-        final String fieldName = fieldType.name();
+        final String fieldName = fieldType.nameForIndex();
```

Comment:
this is an example where i always try and pass the real MappedFieldType even if its an AliasFieldType and let polymorphism pick the right "name" for the index operation.